sdcm/provision/azure/virtual_machine_provider.py

```diff
@@ -142,13 +144,15 @@ def delete(self, name: str, wait: bool = True):
             LOGGER.info("Instance %s has been terminated.", name)
         del self._cache[name]
 
-    def reboot(self, name: str, wait: bool = True) -> None:
+    def reboot(self, name: str, wait: bool = True, hard: bool = False) -> None:
         LOGGER.info("Triggering reboot of instance: %s", name)
-        task = self._azure_service.compute.virtual_machines.begin_restart(self._resource_group_name, vm_name=name)
-        if wait is True:
-            LOGGER.info("Waiting for reboot of instance: %s...", name)
-            task.wait()
-            LOGGER.info("Instance %s has been rebooted.", name)
+        flags = "-ff" if hard else "-f"
+        self.run_command(name, f"reboot {flags}")
```

Comment:
Won't this command hang due to concurrency of closing connection and stopping VM and breaking that connection?

Reply:
I don't know, I tried manually couple of times and was working.
I tried also with -n flag to make it 'harder' :
```
       -n, --no-sync
           Don't sync hard disks/storage media before halt, power-off, reboot.
```
and this was breaking machine for >30 mins (or until I tried to see ssh port status in Azure console?).

Anyway I now try 'multiple hard reboot' nemesis - if that passes I think this solution will be better than current one.